FrontEnd/src/app/components/Products/mylistings/mylistings.component.html

```diff
@@ -2,54 +2,65 @@
 
 <div class="product-list">
 
-  <mat-form-field>
-    <mat-label>Search</mat-label>
-    <input matInput (keyup)="applyFilter($event)" placeholder="Find Product" #input>
-  </mat-form-field>
+  <div class="filter">
+    <mat-form-field class="search-input">
+      <mat-label>Search</mat-label>
+      <input matInput (keyup)="applyFilter($event)" placeholder="Find Product" #input>
+    </mat-form-field>
+    <img routerLink="/uploadProduct" class="add-icon" src="assets/images/plus.png" alt="">
+  </div>
 
-  <table mat-table [dataSource]="allData" class="mat-elevation-z8">
+  <div class="mat-elevation-z8">
+    <table mat-table [dataSource]="dataSource">
 
-    <ng-container matColumnDef="image">
-      <th mat-header-cell *matHeaderCellDef> Cover Image </th>
-      <td mat-cell *matCellDef="let prod"> <img class="prod-img" [src]="getImageUrl(prod.coverImage)" alt="Card image cap"> </td>
-    </ng-container>
-
-    <ng-container matColumnDef="name">
-      <th mat-header-cell *matHeaderCellDef> Name </th>
-      <td mat-cell *matCellDef="let prod"> {{prod.name}} </td>
-    </ng-container>
+      <ng-container matColumnDef="image">
+        <th mat-header-cell *matHeaderCellDef> Cover Image </th>
+        <td mat-cell *matCellDef="let prod"> <img class="prod-img" [src]="getImageUrl(prod.coverImage)" alt="Card image cap"> </td>
+      </ng-container>
   
-    <ng-container matColumnDef="price">
-      <th mat-header-cell *matHeaderCellDef> Price </th>
-      <td mat-cell *matCellDef="let prod"> {{prod.price}} </td>
-    </ng-container>
+      <ng-container matColumnDef="name">
+        <th mat-header-cell *matHeaderCellDef> Name </th>
+        <td mat-cell *matCellDef="let prod"> {{prod.name}} </td>
+      </ng-container>
+    
+      <ng-container matColumnDef="price">
+        <th mat-header-cell *matHeaderCellDef> Price </th>
+        <td mat-cell *matCellDef="let prod"> {{prod.price}} </td>
+      </ng-container>
+    
+      <ng-container matColumnDef="discount">
+        <th mat-header-cell *matHeaderCellDef> Discount </th>
+        <td mat-cell *matCellDef="let prod"> {{prod.discount}} </td>
+      </ng-container>
+    
+      <ng-container matColumnDef="quantity">
+        <th mat-header-cell *matHeaderCellDef> Quantity </th>
+        <td mat-cell *matCellDef="let prod"> {{prod.quantity}} </td>
+      </ng-container>
   
-    <ng-container matColumnDef="discount">
-      <th mat-header-cell *matHeaderCellDef> Discount </th>
-      <td mat-cell *matCellDef="let prod"> {{prod.discount}} </td>
-    </ng-container>
+      <ng-container matColumnDef="description">
+        <th mat-header-cell *matHeaderCellDef> Description </th>
+        <td mat-cell *matCellDef="let prod"> {{prod.description.substring(0, 50)}}{{ prod.description.length > 15 ? '...' : '' }} </td>
```

Comment:
{{ prod.description.length > 15 ? prod.description.substring(0, 50) + '...' : prod.description }}

And Any specific reason for using hard code value 50.